services/eth_gateway.go

```diff
@@ -1024,17 +1049,21 @@ func sendPRLFromOyster(msg OysterCallMsg) (bool, string, int64) {
 		log.Printf("unable to access contract instance at : %v", err)
 	}
 
-	log.Printf("using wallet key store from: %v", msg.From)
+	log.Printf("using wallet key store from: %v", MainWalletAddress)
 	// initialize transactor // may need to move this to a session based transactor
-	auth := bind.NewKeyedTransactor(&msg.PrivateKey)
+	auth := bind.NewKeyedTransactor(MainWalletPrivateKey)
```

Comment:
MainWalletAddress and MainWalletPrivateKey should be changed back to msg.From and &msg.PrivateKey.  We need to send PRLs from other addresses besides just the main wallet.